js/huobi.js

```diff
@@ -5151,41 +5152,90 @@ module.exports = class huobi extends Exchange {
 
     parsePosition (position, market = undefined) {
         //
-        //     {
-        //       symbol: 'BTC',
-        //       contract_code: 'BTC-USDT',
-        //       volume: '1.000000000000000000',
-        //       available: '1.000000000000000000',
-        //       frozen: '0E-18',
-        //       cost_open: '47162.000000000000000000',
-        //       cost_hold: '47151.300000000000000000',
-        //       profit_unreal: '0.007300000000000000',
-        //       profit_rate: '-0.000144183876850008',
-        //       lever_rate: '2',
-        //       position_margin: '23.579300000000000000',
-        //       direction: 'buy',
-        //       profit: '-0.003400000000000000',
-        //       last_price: '47158.6',
-        //       margin_asset: 'USDT',
-        //       margin_mode: 'isolated',
-        //       margin_account: 'BTC-USDT',
-        //       margin_balance: '24.973020070000000000',
-        //       margin_position: '23.579300000000000000',
-        //       margin_frozen: '0',
-        //       margin_available: '1.393720070000000000',
-        //       profit_real: '0E-18',
-        //       risk_rate: '1.044107779705080303',
-        //       withdraw_available: '1.386420070000000000000000000000000000',
-        //       liquidation_price: '22353.229148614609571788',
-        //       adjust_factor: '0.015000000000000000',
-        //       margin_static: '24.965720070000000000'
-        //     }
+        // fetchPositions
+        //
+        //     linear swap (cross & isolated):
+        //
+        //         {
+        //           symbol: 'BTC',
+        //           contract_code: 'BTC-USDT',
+        //           volume: '1.000000000000000000',
+        //           available: '1.000000000000000000',
+        //           frozen: '0E-18',
+        //           cost_open: '47162.000000000000000000',
+        //           cost_hold: '47162.000000000000000000',
+        //           profit_unreal: '0.047300000000000000',
+        //           profit_rate: '0.002005852169119206',
+        //           lever_rate: '2',
+        //           position_margin: '23.604650000000000000',
+        //           direction: 'buy',
+        //           profit: '0.047300000000000000',
+        //           last_price: '47209.3',
+        //           margin_asset: 'USDT',
+        //           margin_mode: 'isolated',
+        //           margin_account: 'BTC-USDT'
+        //           trade_partition: 'USDT',
+        //           position_mode: 'dual_side',
+        //           contract_type: "swap", // present in cross-mode
+        //           pair: "BTT-USDT", // present in cross-mode
+        //           business_type: "swap", // present in cross-mode
+        //         }
+        //
+        //    inverse swap (cross-only, 'unit' mode doesnt affect the response)
```

Comment:
The comments are a bit weird here, huobi has all 4 types of markets actually:

- linear swaps (linear perpetual futures)
- inverse swaps (inverse perpetual futures)
- linear futures (expiring contracts)
- inverse futures (expiring contracts)

Hence it's a bit weird to see the `future` subheader in the comments following the inverse swap header. Technically we should have mentioned all 4 structures here, probably (unless some of those position types are identical).

Reply:
let me revise, i was a bit confused while navigating in their UI, because of having other markets under `futures`.

Reply:
comments fixed and refined for fetchPositions. you can take a look again, they are now more readable and compact